src/main/kotlin/io/github/cottonmc/jsonfactory/builder/JsonFactoryBuilder.kt

```diff
@@ -0,0 +1,96 @@
+package io.github.cottonmc.jsonfactory.builder
+
+import io.github.cottonmc.jsonfactory.data.Identifier
+import io.github.cottonmc.jsonfactory.frontend.Frontend
+import io.github.cottonmc.jsonfactory.frontend.Generator
+import io.github.cottonmc.jsonfactory.frontend.MessageType
+import io.github.cottonmc.jsonfactory.gens.ContentGenerator
+import java.io.File
+
+/**
+ * Builder class, as an API frontend.
+ * */
+class JsonFactoryBuilder() : Frontend {
+    override fun printMessage(msg: String, type: MessageType) {
+        when (type) {
+            MessageType.Error -> error(msg)
+            MessageType.Important -> println(msg)
+            MessageType.Warn -> println(msg)
+            MessageType.Default -> println(msg)
+        }
+    }
+
+    override fun printSeparator() {
+        println()
+    }
+
+    override fun onFinishedGenerating() {
+        printMessage("generation finished", MessageType.Default)
+    }
+
+    override suspend fun shouldOverwriteFile(file: File): Boolean {
+        return false
+    }
+
+    override suspend fun selectOutputDirectory(): File? {
+        return folder
+    }
+
+    var folder: File = File(".")
+    val generators = ArrayList<ContentGenerator>()
+    val identifiers = ArrayList<Identifier>()
```

Comment:
Mutable collections shouldn't be exposed as mutable (or possibly exposed at all).

Reply:
Privatized, I really just wanted to get it working.